viewer/packages/cad-styling/src/prioritized/BoxClusterer.ts

```diff
@@ -29,21 +29,37 @@ export class SmartMergeBoxes implements BoxClusterer {
     this.resultBoxes.pop();
   }
 
-  private shouldMergeBoxesAtIndices(i: number, j: number): boolean {
-    const BOX_MERGE_MIN_IOU = 0.15;
-    const overlap = intersectionOverUnion(this.resultBoxes[i], this.resultBoxes[j]);
-    return overlap >= BOX_MERGE_MIN_IOU;
+  private surfaceArea(box: Box3) {
+    const size = box.getSize(new Vector3());
```

Comment:
Move `new Vector3()` to a pre-allocated var that's reused every call to avoid allocation in this function (like you do with reusableBoundingBox below, but see comments below first).